source/git/gitsource.go

```diff
@@ -175,6 +183,7 @@ func (gs *gitSourceHandler) CacheKey(ctx context.Context, index int) (string, bo
 	defer gs.locker.Unlock(remote)
 
 	if isCommitSHA(ref) {
+		ref = gs.shaToCacheKey(ref)
```

Comment:
ref is used in `ls-remote origin` but it is modified by shaToCacheKey, is that expected?